src/material/slide-toggle/_slide-toggle-theme.scss

```diff
@@ -1,82 +1,116 @@
+@use 'sass:map';
 @use '@material/switch/switch-theme' as mdc-switch-theme;
 @use '@material/form-field' as mdc-form-field;
 @use '../core/theming/theming';
 @use '../core/theming/inspection';
 @use '../core/mdc-helpers/mdc-helpers';
 @use '../core/typography/typography';
-@use '../core/tokens/m2/mdc/switch' as m2-mdc-switch;
-@use '../core/tokens/m2/mat/slide-toggle' as m2-mat-slide-toggle;
+@use '../core/tokens/m2/mdc/switch' as tokens-mdc-switch;
+@use '../core/tokens/m2/mat/slide-toggle' as tokens-mat-slide-toggle;
 @use '../core/tokens/token-utils';
 
 @mixin base($theme) {
-  .mdc-switch {
-    @include mdc-switch-theme.theme(m2-mdc-switch.get-unthemable-tokens());
+  @if inspection.get-theme-version($theme) == 1 {
+    @include _theme-from-tokens(inspection.get-theme-tokens($theme, base));
+  }
+  @else {
+    .mdc-switch {
```

Comment:
Should we target our own class (`.mat-mdc-slide-toggle`) instead of MDC's?

Reply:
I'm not trying to change the M2 styling at all, just adding an alternate path for M3 styling. There is currently no `mat-` class on that element, so leaving it as is.